### PR TITLE
Admin log view

### PR DESCRIPTION
**Pull request by juliangiebel**
First version of the admin log view.
I will keep working on this but I'm opening a PR so I can get a review before I add more stuff and so it can be used already.

The links in the details pane will point to overview pages for the server/round/player when I implement them.

I'm not a designer so I would appreciate suggestions on how to make this look and feel better.

Fulltextsearch doesn't work right now because the index isn't being used properly but that needs to be fixed in content.

### Screenshots:
grafik
grafik

### CODE REVIEW

**Comment by PJB3005** [comment]
VERY based

File: SS14.Admin/Pages/Logs/Index.cshtml.cs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+        public LogsIndexModel(PostgresServerDbContext dbContext)
+        {
+            _dbContext = dbContext;
+        }
+
+        public async Task OnGetAsync(string? sort, int? pageIndex, int? perPage)
+        {
+            if (FromDate != DateTime.MinValue) AllRouteData.Add("fromDate", FromDate.ToString("yyyy-MM-dd"));
+            if (FromDate != DateTime.MaxValue) AllRouteData.Add("toDate", ToDate.ToString("yyyy-MM-dd"));
```
Comment (DrSmugleaf) [inline SS14.Admin/Pages/Logs/Index.cshtml.cs:38]:
```suggestion
            if (FromDate != default) AllRouteData.Add("fromDate", FromDate.ToString("yyyy-MM-dd"));
            if (ToDate != default) AllRouteData.Add("toDate", ToDate.ToString("yyyy-MM-dd"));
```

**Review by DrSmugleaf — CHANGES_REQUESTED** [review]

File: SS14.Admin/AdminLogs/LogJsonTagHelper.cs [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        return result != default;
+    }
+
+    private async Task<string> TranslateProperty(JsonProperty property) => property.Name switch
+    {
+        "player" => await RetrievePlayerLink(property),
+        _ => $"<p>{property.Name}: {property.Value.GetRawText() ?? "NULL"}</p>"
+    };
+
+    private string ParsePositon(JsonProperty property)
```
Comment (DrSmugleaf) [inline SS14.Admin/AdminLogs/LogJsonTagHelper.cs:86]:
```suggestion
            "coordinates" => ParsePosition(entry),
            "position" => ParsePosition(entry),
            _ => default
        };

        return result != default;
    }

    private async Task<string> TranslateProperty(JsonProperty property) => property.Name switch
    {
        "player" => await RetrievePlayerLink(property),
        _ => $"<p>{property.Name}: {property.Value.GetRawText() ?? "NULL"}</p>"
    };

    private string ParsePosition(JsonProperty property)

```

File: SS14.Admin/Pages/Logs/Index.cshtml.cs [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+            SortState.AddColumn("date", entry => entry.Date, SortOrder.Descending);
+            SortState.AddColumn("impact", entry => entry.Impact);
+            SortState.AddColumn("type", entry => entry.Type);
+            SortState.AddColumn("message", entry => entry.Message);
+            SortState.Init(sort, AllRouteData);
+
+            Pagination.Init(pageIndex, perPage, AllRouteData);
+
+            //.Include(log => log.Players)
+            //.ThenInclude(player => player.Player)
```
Comment (DrSmugleaf) [inline SS14.Admin/Pages/Logs/Index.cshtml.cs:50]: Are these supposed to be included?

**Review by DrSmugleaf — CHANGES_REQUESTED** [review]

File: SS14.Admin/Models/JsonQueryBinder.cs [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+                }
+            }
+            catch (JsonException e)
+            {
+                bindingContext.ActionContext.ModelState.TryAddModelError(e.Path ?? string.Empty, e,
+                    bindingContext.ModelMetadata);
+            }
+            catch (Exception e) when (e is FormatException or OverflowException)
+            {
+                Console.WriteLine(e);
```
Comment (PJB3005) [inline SS14.Admin/Models/JsonQueryBinder.cs:55]: Probably don't want to keep this console writeline.

File: SS14.Admin/Pages/Logs/Index.cshtml.cs (hunk 3 above)
Comment (juliangiebel) [reply SS14.Admin/Pages/Logs/Index.cshtml.cs]: No. I removed the comments.

File: SS14.Admin/Pages/Logs/Index.cshtml.cs (hunk 1 above)
Comment (juliangiebel) [reply SS14.Admin/Pages/Logs/Index.cshtml.cs]: I changed FromDate to ToDate on line 38

**Pull request merged**